agenta-backend/agenta_backend/migrations/mongo_to_postgres/migration.py

```diff
@@ -0,0 +1,548 @@
+        TemplateType(template["type"]) if "type" in template else TemplateType.IMAGE
+    )
+
+    return {
+        "id": template_uuid,
+        "type": template_type,
+        "template_uri": template.get("template_uri"),
+        "tag_id": template.get("tag_id"),
+        "name": template["name"],
+        "repo_name": template.get("repo_name"),
+        "title": template["title"],
+        "description": template["description"],
+        "size": template.get("size"),
+        "digest": template.get("digest"),
+        "last_pushed": get_datetime(template.get("last_pushed")),
+    }
+
+
+async def transform_test_set(test_set):
+    app_uuid = await get_mapped_uuid("app_db", test_set["app"].id)
+    user_uuid = await get_mapped_uuid("users", test_set["user"].id)
+    test_set_uuid = generate_uuid()
+    await store_mapping("testsets", test_set["_id"], test_set_uuid)
+    return {
+        "id": test_set_uuid,
+        "name": test_set["name"],
+        "app_id": app_uuid,
+        "csvdata": test_set["csvdata"],
+        "user_id": user_uuid,
+        "created_at": get_datetime(test_set.get("created_at")),
+        "updated_at": get_datetime(test_set.get("updated_at")),
+    }
+
+
+async def transform_evaluator_config(config):
+    app_uuid = await get_mapped_uuid("app_db", config["app"].id)
+    user_uuid = await get_mapped_uuid("users", config["user"].id)
+    config_uuid = generate_uuid()
+    await store_mapping("evaluators_configs", config["_id"], config_uuid)
+    return {
+        "id": config_uuid,
+        "app_id": app_uuid,
+        "user_id": user_uuid,
+        "name": config["name"],
+        "evaluator_key": config["evaluator_key"],
+        "settings_values": config["settings_values"],
+        "created_at": get_datetime(config.get("created_at")),
+        "updated_at": get_datetime(config.get("updated_at")),
+    }
+
+
+async def convert_human_evaluations_associated_variants(
+    variants, variants_revisions, evaluation_id
+):
+    """Convert variant and revision ObjectIds to UUIDs and structure them."""
+    associated_variants = []
+    assert len(variants) == len(
+        variants_revisions
+    ), "variants and variants_revisions must have the same length"
+
+    for variant_id, revision_id in zip(variants, variants_revisions):
+        variant_uuid = await get_mapped_uuid("app_variants", variant_id)
+        revision_uuid = await get_mapped_uuid("app_variant_revisions", revision_id)
+        associated_variants.append(
+            {
+                "human_evaluation_id": evaluation_id,
+                "variant_id": variant_uuid,
+                "variant_revision_id": revision_uuid,
+            }
+        )
+    return associated_variants
+
+
+async def transform_human_evaluation(evaluation):
+    app_uuid = await get_mapped_uuid("app_db", evaluation["app"].id)
+    user_uuid = await get_mapped_uuid("users", evaluation["user"].id)
+    test_set_uuid = await get_mapped_uuid("testsets", evaluation["testset"].id)
```

Comment:
Expect the test set to be deleted.

Reply:
if the testset is deleted the foreign key will be null. what's the matter there ?